### PR TITLE
One quiz

### PR DESCRIPTION
**Pull request by cindy1u0**
Show one quiz -> to test it need to change the url to /assessments/<number> and then click on the title
1. I couldn't really get the selection work because i tried using onclick but every time i click it then the pictures rearrange...

I also refactored some typescript files and then moved all the @types to devdependencies @stefan-philip lmk what u think or if u find anything wrong with it

... so TLDR i factored a lot shit click around if anything breaks lmk... or if i put up debugging message that i dont rmb to remove... thanks guys <333

### CODE REVIEW

File: frontend/src/components/PurpleButton.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  }
+`;
+
+const PurpleButton = ({ text='', top=0, onClick={} }) => {
+  return (
+    // @ts-ignore
+    <ButtonContainer top={top} onClick={onClick}>
+      {text}
+    </ButtonContainer>
+  );
```
Comment (jrosen081) [inline frontend/src/components/PurpleButton.tsx]: Why does this ignore ts?

File: frontend/src/pages/Assessments/Quiz.tsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+const connector = connect(
+  (state) => ({
+    allWords: getAllWords(state),
+  }),
+  {
+    getWords: getWords.request,
+  }
+);
+
+const shuffleImages = (images: any) => {
```
Comment (jrosen081) [inline frontend/src/pages/Assessments/Quiz.tsx]: Wouldn't this be a `any[]`?

File: frontend/src/pages/Assessments/Quiz.tsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+import Layout from "../../components/Layout";
+import { getWords } from "./data/actions";
+import { getAllWords } from "./data/reducer";
+import styled from "styled-components";
+import PromptSpeech from "../../components/PromptSpeech";
+import PurpleButton from "../../components/PurpleButton";
+import { Word } from "../../models/types";
+import { CLOUD } from "../../constants/colors";
+
+interface IProps {
```
Comment (jrosen081) [inline frontend/src/pages/Assessments/Quiz.tsx]: Can we call this `QuizProps`?

File: frontend/src/pages/Assessments/Quiz.tsx [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+  getWords: Function,
```
Comment (jrosen081) [inline frontend/src/pages/Assessments/Quiz.tsx]: Can we specify what type the function is? Like (type: any) => void;?

File: frontend/src/pages/Assessments/Quiz.tsx [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+  allWords: Array<Word>,
```
Comment (jrosen081) [inline frontend/src/pages/Assessments/Quiz.tsx]: Didn't know you could do it like this, I have been doing  Word[] :)

**Review by jrosen081 — COMMENTED** [review]
This was also a chungus lol

File: frontend/src/pages/Assessments/Quiz.tsx (hunk 5 above)
Comment (cindy1u0) [reply frontend/src/pages/Assessments/Quiz.tsx]: Oh i think maybe doing Word[] is prob better lol im not super familiar with typescript rip ill chagne this

File: frontend/src/pages/Assessments/Quiz.tsx (hunk 4 above)
Comment (cindy1u0) [reply frontend/src/pages/Assessments/Quiz.tsx]: okie

File: frontend/src/pages/Assessments/Quiz.tsx (hunk 3 above)
Comment (cindy1u0) [reply frontend/src/pages/Assessments/Quiz.tsx]: aight ill change this

File: frontend/src/components/PurpleButton.tsx (hunk 1 above)
Comment (cindy1u0) [reply frontend/src/components/PurpleButton.tsx]: oh wait nvm i think i added it before i fixed the problem. I will change that thanks for catching it

**Review by stefan-philip — APPROVED** [review]

**Pull request merged**